azure-pipelines.yml

```diff
@@ -352,8 +352,12 @@ stages:
 
         mkdir dist
         pip wheel --no-build-isolation -v -v -v --wheel-dir=dist .
-        ls dist -r | Foreach-Object {
-            pip install $_.FullName
+        If ($lastExitCode -ne "0") {
+            Write-Host "##vso[task.complete result=Failed;]DONE"
```

Comment:
Wouldn't it be simpler to `Exit 1` here and let Azure detect that the script exited with a non-zero status and fail the task that way (assuming it works)?

I have also used `ErrorActionPreference` in the past, like [here](https://github.com/pyvista/gl-ci-helpers/blob/86c492437b68139e3d7a2973f09a26950758aa8d/azure-pipelines.yml#L28-L30). This might be better as it would work script-wide for each step.

Not 100% sure if either of these works. I'd be inclined to try the `ErrorActionPreference` one and see if you can make it fail (for example by changing the line above temporarily to `pip wheel --foo ...` to force it to fail), but I can try that in a separate PR if you want to get this in as-is!

Reply:
Probably. This `##vso` came from an issue I saw the multiline problem discussed in, and it was already used in this file so I used it for consistency.

> I have also used `ErrorActionPreference` in the past, like [here](https://github.com/pyvista/gl-ci-helpers/blob/86c492437b68139e3d7a2973f09a26950758aa8d/azure-pipelines.yml#L28-L30). This might be better as it would work script-wide for each step.


Does it requires all these 3 lines?
```
      Set-StrictMode -Version Latest
      $ErrorActionPreference = "Stop"
      $PSDefaultParameterValues['*:ErrorAction']='Stop'
```
Your example is still for a single `powershell: |` step - if it's no more global than that, then I think it's probably the worst of the three options, due to verbosity and no actual gain (there's only two commands that can fail here, and the second one is already working with the regular exit status check.

Reply:
> if it's no more global than that, then I think it's probably the worst of the three options, due to verbosity and no actual gain (there's only two commands that can fail here, and the second one is already working with the regular exit status check.

While it's true that the effect here is minimal (only would check one additional line) the current solution is not very generalizable -- as you add more commands the current solution requires adding more and more exit value checks and remains Azure-specific (using Azure magic rather than standard return values). Thus I still prefer something like the equivalent of `set -e` in the `bash: |` lines (also for consistency with that in bash!), but I can live with the current solution.